### PR TITLE
WM4/OlivierSedzikowski/JS-core2-week3

### PR DESCRIPTION
**Pull request by OlivierSedzikowski**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Olivier Sedzikowski
- Your City: Birmingham
- Your Slack Name: oliviersezikowski

# Homework Details

- Module: JS-core 2
- Week: 3

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?
Need my code to be review.

### CODE REVIEW

**Comment by OlivierSedzikowski** [comment]
I started excercise-b but I stuck in line 6 (instructions).

File: 1-exercises/B-highlight-words/script.js [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+// - When clicked, we need to check the value of the `<select>` element using the `.value` property.
+// - We can then update the `background-color` property of the `<span>` with the value of the select - remember that the value "none" is a special case and we need to be handled differently.
+// - All of your HTML should go inside the `<div>` with the id **"content"**.
+
 function highlightWords(paragraph, colours) {
-  // Write your code here...
+for (let a = 0; a < array.length; a++) {
+  let paragraph = document.createElement("p");
+  paragraph.innerHTML = paragraph;
+  paragraph.split("") = array;
```
Comment (magarpratik) [inline 1-exercises/B-highlight-words/script.js:16]:
```suggestion
  paragraph.split(" ") = array;
```

Comment (magarpratik) [reply 1-exercises/B-highlight-words/script.js:16]: The argument of `.split()` should be `" "` as we want to split the paragraph based on the spaces between the words.

File: 1-exercises/B-highlight-words/script.js [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+// - For each word, create a `<span>` element and set the innerText to the word, plus a space - " ". Add this to the `<p>`. <<<???>>>
+// - Each `<span>` should have an eventListener that will listen for clicks.
+// - When clicked, we need to check the value of the `<select>` element using the `.value` property.
+// - We can then update the `background-color` property of the `<span>` with the value of the select - remember that the value "none" is a special case and we need to be handled differently.
+// - All of your HTML should go inside the `<div>` with the id **"content"**.
+
 function highlightWords(paragraph, colours) {
-  // Write your code here...
+for (let a = 0; a < array.length; a++) {
+  let paragraph = document.createElement("p");
```
Comment (magarpratik) [inline 1-exercises/B-highlight-words/script.js:14]: This line should be outside of the for loop.

File: 1-exercises/B-highlight-words/script.js [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
+// - Iterate over the array of words.
+// - For each word, create a `<span>` element and set the innerText to the word, plus a space - " ". Add this to the `<p>`. <<<???>>>
+// - Each `<span>` should have an eventListener that will listen for clicks.
+// - When clicked, we need to check the value of the `<select>` element using the `.value` property.
+// - We can then update the `background-color` property of the `<span>` with the value of the select - remember that the value "none" is a special case and we need to be handled differently.
+// - All of your HTML should go inside the `<div>` with the id **"content"**.
+
 function highlightWords(paragraph, colours) {
-  // Write your code here...
+for (let a = 0; a < array.length; a++) {
```
Comment (magarpratik) [inline 1-exercises/B-highlight-words/script.js:13]: You don't need this for loop.

**Pull request closed without merging**